File: slack-poster/src/main/java/dev/number6/slackposter/model/PresentableChannelSummary.java
package dev.number6.slackposter.model;

import dev.number6.slackposter.ChannelSummaryImage;

import java.text.MessageFormat;

public class PresentableChannelSummary {

    //\u2022 == bullet point character
    public static final String CHANNEL_MESSAGE_TOTAL_FORMAT = "\u2022 *{0}*: {1} messages on {2}";

    private final ChannelSummaryImage image;

    public PresentableChannelSummary(ChannelSummaryImage image) {
        this.image = image;
    }

    public String getInitialMessageLine() {

        return MessageFormat.format(CHANNEL_MESSAGE_TOTAL_FORMAT,
                image.getChannelName(),
                image.getSentimentTotals().values().stream().reduce(0, Integer::sum),
                image.getComprehensionDate());
    }

    public Attachment[] getAttachments() {
        return new Attachment[]{getOverallSentimentAttachment(), getEntityAttachment(), getKeyPhrasesAttachment()};
    }

    private Attachment getOverallSentimentAttachment() {
        return new SentimentAttachment(image);
    }

    private Attachment getKeyPhrasesAttachment() {
        return new KeyPhrasesAttachment(image);
    }

    private Attachment getEntityAttachment() {
        return new EntityAttachment(image);
    }
}

File: slack-poster/src/main/java/dev/number6/slackposter/model/PresentableChannelSummary.kt
package dev.number6.slackposter.model

import dev.number6.slackposter.ChannelSummaryImage
import java.text.MessageFormat

class PresentableChannelSummary(private val image: ChannelSummaryImage) {
    val initialMessageLine: String
        get() = MessageFormat.format(CHANNEL_MESSAGE_TOTAL_FORMAT,
                image.channelName,
                image.sentimentTotals.values.stream().reduce(0) { a: Int, b: Int -> Integer.sum(a, b) },
                image.comprehensionDate)

    val attachments: Array<Attachment>
        get() = arrayOf(overallSentimentAttachment, entityAttachment, keyPhrasesAttachment)

    private val overallSentimentAttachment: Attachment
        get() = SentimentAttachment(image)

    private val keyPhrasesAttachment: Attachment
        get() = KeyPhrasesAttachment(image)

    private val entityAttachment: Attachment
        get() = EntityAttachment(image)

    companion object {
        //\u2022 == bullet point character
        const val CHANNEL_MESSAGE_TOTAL_FORMAT = "\u2022 *{0}*: {1} messages on {2}"
    }

}